[AIRFLOW-480] Allow for binary file download from Google Cloud Storage

volkale opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:

https://issues.apache.org/jira/browse/AIRFLOW-480
Per Apache guidelines you need to create a Jira issue.

Testing Done:

This bug fix allowed me to download a gzip file from GCS.

volkale committed:
[AIRFLOW-480] Allow for binary file download from Google Cloud Storage

volkale committed:
[AIRFLOW-480] Allow for binary file download from Google Cloud Storage

Commit:
[AIRFLOW-483] Change print to logging statement

Dear Airflow Maintainers,

Please accept this PR that addresses the following
issues:

https://issues.apache.org/jira/browse/AIRFLOW-483

Testing Done:

This fix prevented the stdout from being spammed
by the file content.

Closes #1780 from
skogsbaeck/fix/gcs_download_operator

alexvanboxel committed:
[AIRFLOW-467] Allow defining of project_id in BigQueryHook

Introduced a toplevel table splitter that also
allows for defining a project_id as a suffix seperated by
the legacy colon or the new dot. If the project is not
defined, the default project_id will be used.

As it's so common of all the BigQuery classes
(Hook, Cursor, ...) the same splitter is used over all
those classes.

The documentation is adapted to allow defining the
suffix seperated as with the legacy colon as well
as with the new SQL dotted notation.

The change is 100% backwards compatible. Unit
tests are added for all scenario's, including negative
and compatibility.

Closes #1781 from alexvanboxel/feature/bq_tablename

alexvanboxel committed:
[AIRFLOW-477][AIRFLOW-478] Restructure security section for clarity

Closes #1775 from alexvanboxel/docs/security

alexvanboxel committed:
[AIRFLOW-159] Add cloud integration section + GCP documentation

Closes #1773 from alexvanboxel/feature/gcp-docs

alexvanboxel committed:
[AIRFLOW-468] Update Panda requirement to 0.17.1

BigQuery Hook requires at least Panda 0.17.1

Closes #1767 from alexvanboxel/feature/panda-
upgrade

forevernull committed:
[AIRFLOW-488] Fix test_simple fail

Make unittest test_simple pass on all platforms
including MacOs.

Closes #1782 from forevernull/master

bolkedebruin committed:
Revert "[AIRFLOW-78] airflow clear leaves dag_runs"

This reverts commit 197c9050ef3a142c18aa97819da48ee8cadbf8d8.

Regressions were observed and tasks were not scheduled in case of
max_dag_runs reached.

criccomini commented on airflow/contrib/hooks/gcs_hook.py:
```diff
@@ -59,7 +59,10 @@ def download(self, bucket, object, filename=False):
         # Write the file to local file path, if requested.
         if filename:
-            with open(filename, 'w') as file_fd:
+            write_argument = 'w'
```
Pythonic:

```
write_argument = 'wb' if isinstance(downloaded_file_bytes, bytes) else 'w'
```

aoen committed:
[AIRFLOW-494] Add per-operator success/failure metrics

Adds metrics for success/failure rates of each operator, that way
when we e.g. do a new release we will have some
signal if there is a regression in an operator. It
will also be useful if e.g. a user wants to
upgrade their infrastructure and make sure that
all of the operators still work as expected.

Testing Done:
- Local staging and make sure that several
operators successes/failures were accurately
reflected

Closes #1785 from aoen/ddavydov/add_per_operator_s
uccess_fail_metrics

dud225 committed:
[AIRFLOW-469] Add MFG Labs as Airflow user

Hello.

We'd like to be added as official Airflow users.

Regards

Closes #1768 from dud225/patch-1

volkale committed:
Merge branch 'feature/allow_binary_input' of github.com:markovianhq/incubator-airflow into feature/allow_binary_input

volkale committed:
[AIRFLOW-480] Make code more Pythonic
